crates/common_utils/src/crypto.rs

```diff
@@ -215,6 +215,23 @@ impl DecodeMessage for GcmAes256 {
     }
 }
 
+/// Secure Hash Algorithm 512
+#[derive(Debug)]
+pub struct Sha512;
+
+/// Trait for generating a digest for SHA
+pub trait GenerateDigest {
+    /// takes a message and creates a digest for it
+    fn generate_digest(&self, message: &[u8]) -> CustomResult<Vec<u8>, errors::CryptoError>;
```

Comment:
can we have the output of this function as `&[u8]`?

Reply:
Cannot return a reference of owned value from a function!